content/blog/2025-05-27-introducing-enhanced-json-capabilities-in-valkey.md

```diff
@@ -0,0 +1,182 @@
+
+JSON has become the universal format for data exchange, but effectively managing JSON data at scale remains a challenge in caching layers. The Valkey-JSON module addresses this by providing a dedicated JSON data type and a rich set of commands for inserting, updating, and retrieving JSON content using JSONPath queries. This approach reduces application complexity and accelerates performance, making it ideal for modern, data-driven applications.
+
+Let’s first cover the key capabilities and performance of Valkey-JSON. We will then dive into how to work with Valkey-JSON and demonstrate how to get started with a real world example. 
+
+## Core Capabilities and Performance
+
+Valkey’s JSON support goes beyond basic storage and retrieval. It introduces powerful querying and filtering capabilities that streamline complex data operations. Tasks that once required extensive development time and multiple network requests can now be executed through a single JSON command, significantly enhancing developer productivity and reducing operational costs.
+
+Valkey-JSON supports six data types—**null, boolean, number, string, object, and array**—allowing developers to represent complex, nested data structures without the constraints of traditional string serialization. Unlike other composite types, JSON objects and arrays in Valkey can contain any combination of the six value types, enabling deeply nested and recursive data models to be stored natively.
+
+Internally, Valkey-JSON utilizes an optimized **binary tree-like format**, which enables rapid traversal and manipulation of substructures without requiring full document parsing. This structure not only minimizes memory usage but also ensures that operations on specific paths remain efficient. Path-based commands like `JSON.GET`, `JSON.SET`, and `JSON.DEL` allow targeted interactions with specific elements, supporting multiple paths within a single operation. Additionally, Valkey-JSON integrates with Valkey’s Access Control Lists (ACLs), introducing a `@json` command category to enforce granular permissions alongside existing data types.
+
+##  Getting Started with Valkey-JSON
+
+Let’s now walk through a real-world example and see how simple, yet powerful Valkey-JSON is to use. 
+
+### Installation and Setup
+
+Valkey-JSON is a module, so you’ll need to first build locally:
```

Comment:
We're going to have a binary distribution right? Maybe not today, but we don't expect people to build it themselves forever.

If this is a temporary thing, I would say something like "At time of writing, you'll need to build the module locally"